api/app/controllers/spree/api/checkouts_controller.rb

```diff
@@ -52,7 +52,7 @@ def update
         authorize! :update, @order, order_token
 
         if @order.update_from_params(params, permitted_checkout_attributes, request.headers.env)
-          if current_api_user.has_spree_role?('admin') && user_id.present?
+          if can?(:admin, @order) && user_id.present?
```

Comment:
I think this is the reason for the `:admin` pseudo-action. All users can edit/update their own order, so admin is to distinguish between that kind of editing and this kind. A good clarification of this somewhere would of course be welcome.